Multi bar

Koleok opened the pull request:
Supporting multi-series bar charts _(and possibly other multi series types)_

Koleok committed:
Add utils

Koleok committed:
Working multibar shaper

ntietz reviewed (commented):
Currently, the bars still show as overlapping when I run this on my local machine.

Koleok committed:
Shaping for multi-series is correct

Datapoints still include __typename keys, and post-formatting
datapoints still include their seriesKey, this shouldn't impede
anything so i left them in rather than stripping.

Something in d3 or the wrapper is still throwing
- TypeError: Cannot add property series, object is not extensible
When i pass in the new multi-series shape, not sure why, so this
is currently my problem at hand.

Koleok committed:
Add some clarifications

Koleok committed:
Use nicks fix for object extensibility

Koleok committed:
Remove log

Koleok committed:
Reverted groupByProp

ntietz committed:
Fix multi bar charts

ntietz committed:
Merge branch 'multi-bar' of github.com:iSCAD-plus/plots-prototype into multi-bar

ntietz committed:
Fill in missing elements of numeric x ranges

Koleok committed:
Update test

Koleok committed:
Slight refactor, did an oop thing 😱
